src/app/backend/handler/apihandler.go

```diff
@@ -1391,10 +1400,15 @@ func (apiHandler *APIHandler) handleGetPodDetail(request *restful.Request, respo
 		handleInternalError(response, err)
 		return
 	}
+	config, err := apiHandler.cManager.Config(nil)
+	if err != nil {
+		handleInternalError(response, err)
+		return
+	}
 
 	namespace := request.PathParameter("namespace")
 	name := request.PathParameter("pod")
-	result, err := pod.GetPodDetail(k8sClient, apiHandler.iManager.Metric().Client(), namespace, name)
+	result, err := pod.GetPodDetail(k8sClient, apiHandler.iManager.Metric().Client(), namespace, name, config.Host)
```

Comment:
Were you able to test it not on a localhost?

Reply:
not sure I understand the question... Are you asking if I was not able to test it on localhost?

Reply:
I meant setup different than localhost, i.e. AWS, GCE etc.

Reply:
Tested running master on a vagrant vm. Dashboard did generated correct links.

Reply:
I think it will be difficult to retrive the correct IP that works on-premise and in the cloud. Actually the node has an external ip field (as far as I remember it is set in GKE), however in many cases it is probably not set (local cluster). 


```
 kubectl get node kube-master -o wide
NAME          STATUS    AGE       VERSION   EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION
kube-master   Ready     11d       v1.7.3    <none>        Ubuntu 16.04.3 LTS   4.4.0-93-generic

```

So, I think it is better not to use it and rather cut of the host in the front end